src/create-pr.js

```diff
@@ -0,0 +1,272 @@
+ *
+ * @returns {string} repository name
+ */
+function getRepoName () {
+  const payload = github.context.payload;
+
+  return payload.repository.name;
+}
+
+/**
+ * Get Last Release object as a promise
+ *
+ * @returns {Promise} last release object
+ */
+async function getLastRelease () {
+  try {
+    var release = await octokit.repos.getLatestRelease({
+      owner: owner,
+      repo: repo
+    });
+
+    return release.data;
+  } catch (e) {
+    return;
+  }
+}
+
+/**
+ * Get all merge Commits between master and sent tag
+ *
+ * @param {string} tagName name of the tag from last release
+ *
+ * @returns {Promise} list of commits
+ */
+async function getMergeCommitsSince (tagName) {
+  const cwd = GITHUB_WORKSPACE + '/' + repo;
+
+  var commits = await simpleGit(cwd)
+    .log({ from: BASE_BRANCH, to: tagName, '--merges': true });
+
+  return commits.all;
+}
+
+/**
+ * Get all merge Commits of the repo
+ *
+ * @returns {Promise} list of commits
+ */
+async function getAllMergeCommitsSinceBeginning () {
+  const cwd = GITHUB_WORKSPACE + '/' + repo;
+
+  var commits = await simpleGit(cwd)
+    .log({ from: BASE_BRANCH, '--merges': true });
+
+  // limiting results to 500 prs, as its unlikely to have more prs in a single
+  // release. Also if the number is high, it crosses the github api request limit,
+  // hence enforcing this limit
+  return commits.all.slice(0, 500);
+}
+
+/**
+ * Get all pull requests
+ * If a release exists, it returns PRs merged since last release
+ * Otherwise it return all merged PRs
+ *
+ * @param {object} lastRelease last release object
+ *
+ * @returns {Promise} list of pull requests
+ */
+async function getPullRequests (lastRelease) {
+  let commits;
+
+  var lastRelease = { tag_name: '1.9.9'};
+  if (lastRelease) {
+    commits = await getMergeCommitsSince(lastRelease.tag_name);
+  } else {
+    commits = await getAllMergeCommitsSinceBeginning()
+  }
+
+  return await fetchPrsForCommits(commits);
+}
+
+/**
+ * Fetch all prs for the merge commits
+ *
+ * @param {Array} commits list of commits
+ * @returns {Promise} list of pull requests
+ */
+async function fetchPrsForCommits (commits) {
+  var promises = await _.map(commits, async function (commit) {
+    var url = 'GET /repos/' + owner + '/' + repo + '/commits/' + commit.hash + '/pulls';
+
+    var data = await octokit.request(url, {
+      owner: owner,
+      repo: repo,
+      commit_sha: commit.sha,
+      mediaType: { previews: ['groot'] }
+    });
+
+    return data.data;
+  });
+
+  var pullRequests = await Promise.all(promises);
+
+  return _.chain(pullRequests)
+    .flatten()
+    .filter(function (pr) {
+      var ifBelongsToCurrentRepo = pr.base.repo.full_name === owner + '/' + repo;
+      var ifPRToMaster = pr.base.ref === BASE_BRANCH;
+      var isNotAReleasePr = !_.find(pr.labels, { 'name': RELEASE_PR_IDENTIFIER_LABEL });
+
+      return ifBelongsToCurrentRepo && ifPRToMaster && isNotAReleasePr;
+    })
+    .map(function (pr) {
```

Comment:
Do we need to filter out duplicate PR's? or this is not a possibility?

Reply:
![2021-03-03 at 4 57 PM](https://user-images.githubusercontent.com/5058867/109799113-9f2a4c80-7c41-11eb-9163-3aee222ff755.png)
As discussed, added a warning message so that user can review and make changed accordingly.